packages/babel-traverse/src/path/evaluation.ts

```diff
@@ -85,35 +84,36 @@ function _evaluate(path, state) {
     path.isNumericLiteral() ||
     path.isBooleanLiteral()
   ) {
-    return node.value;
+    return path.node.value;
   }
 
   if (path.isNullLiteral()) {
     return null;
   }
 
   if (path.isTemplateLiteral()) {
-    return evaluateQuasis(path, node.quasis, state);
+    return evaluateQuasis(path, path.node.quasis, state);
   }
 
   if (
     path.isTaggedTemplateExpression() &&
     path.get("tag").isMemberExpression()
   ) {
-    const object = path.get("tag.object");
+    const object = path.get("tag.object") as NodePath;
     const {
+      // @ts-expect-error todo(flow->ts): possible bug, object is can be any expression and so name might be undefined
       node: { name },
     } = object;
-    const property = path.get("tag.property");
+    const property = path.get("tag.property") as NodePath;
 
     if (
       object.isIdentifier() &&
       name === "String" &&
-      !path.scope.getBinding(name, true) &&
```

Comment:
I doubt it should have been `hasBinding(name, true)`, as the binding is never used later. Can you leave a todo note here?

Reply:
sure, added a comment
sounds like - this might be the case